infrastructure/src/main/java/org/corfudb/infrastructure/BatchWriter.java

```diff
@@ -45,24 +45,22 @@
                     .build());
 
     /**
-     * The epochWaterMark is the epoch up to which all operations have been sealed. Any
-     * BatchWriterOperation arriving after the epochWaterMark with an epoch less than the epochWaterMark, is
-     * completed exceptionally with a WrongEpochException.
+     * The sealEpoch is the epoch up to which all operations have been sealed. Any
+     * BatchWriterOperation arriving after the sealEpoch with an epoch less than the sealEpoch
+     * is completed exceptionally with a WrongEpochException.
      * This is persisted in the ServerContext by the LogUnitServer to withstand restarts.
      */
-    private volatile long epochWaterMark;
+    private volatile long sealEpoch;
```

Comment:
I know you didn't write this, but just an fyi this doesnt need to be a volatile.